docs/backend.md

```diff
@@ -139,3 +139,36 @@ class MyController < ApplicationController
   end
 end
 ```
+
+## Database
+
+We make use of ActiveRecord for our ORM. It is important to remain mindful of what your
+code is doing to prevent unnecessary queries. For example, the following conditional
+should be rearranged:
+
+```ruby
+# Bad: potentially-unnecessary DB query
+if mfa_user.two_factor_enabled? && !in_mfa_selection_flow
```

Comment:
I think this example relies on specific knowledge of the `MfaUser` class which itself wraps an ActiveRecord object.

Maybe we could use some example illustrative names?

```suggestion
if model.association.count > 2 && !local_boolean
```
and then it might be clearer?

Reply:
I debated this a bit, but you're probably right.

Part of what I wanted to try to convey is that `two_factor_enabled?` doesn't tell you it's going to do a database query; you need to pay some attention to what the methods are doing. But maybe this is too obtuse an example.